core/src/main/java/com/graphhopper/routing/Path.java

```diff
@@ -377,7 +377,7 @@ public InstructionList calcInstructions(final Translation tr) {
      * @return List of PathDetails for this Path
      */
     public Map<String, List<PathDetail>> calcDetails(final PathDetailsBuilderFactory pathBuilderFactory, int previousIndex) {
-        if (pathBuilderFactory == null)
+        if (!isFound() || edgeIds.isEmpty() || pathBuilderFactory == null)
```

Comment:
Also understand this now ;) and will add an integration test.

I think the `edgeIds.isEmpty()` should not be here as a PathDetail could still be included even if the path has no length. So we have to do similar things that we do in calcInstructions when the path is empty but the finish instruction should be there.